packages/legend-application-studio/src/stores/editor-state/element-editor-state/ClassState.ts

```diff
@@ -126,7 +126,6 @@ export class DerivedPropertyState extends LambdaEditorState {
             ? this.extractLambdaString(grammarText)
             : '',
         );
-        this.clearErrors();
```

Comment:
I think this is intended behavior, because we change the rendering style from `standard` to `pretty` so source information from the old error might be off.

Reply:
should we recomplile here if there is an existing error ? that way the user keeps the error with the updated source information?

Reply:
I don't think the simple expand/collapse should bear the cost of recompilation. Also, it might be weird if we expand/collapse when user hasn't compile anytime, they will be caught off surprise to know this toggling cause compilation to occur. So IMO, we should leave this as is

Reply:
The issue tho is that when there is an exisiting compilation error on that lambda and they expand it disappears. 
Not suggesting we do a compile everytime they toggle expand (that would be too expensive). but more: 
if there is an existing compile error on that lambda then recompile so we get the updated source information and show the correct source information.
Or maybe if we don't want to recompile, just keep the error message but remove the source info pointers as they are no longer relevant, if the user wants the updated, then they would need to hit f9 again.